### PR TITLE
Lint examples, add editorconfig linting

### LINKED ISSUE #112: 'npm run test' fails on new app
Reproduce:
```
preact create app
cd app/
npm run test
```

```
> app@1.0.0 test /home/oren/app
> eslint src && preact test

The react/jsx-space-before-closing rule is deprecated. Please use the react/jsx-tag-spacing rule with the "beforeSelfClosing" option instead.

/home/oren/app/src/components/app.js
  11:2  error  Expected line before comment  lines-around-comment

/home/oren/app/src/components/header/index.js
  5:16  warning  Component should be written as a pure function  react/prefer-stateless-function

/home/oren/app/src/routes/home/index.js
  4:16  warning  Component should be written as a pure function  react/prefer-stateless-function

✖ 3 problems (1 error, 2 warnings)


npm ERR! Linux 4.4.0-78-generic
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "run" "test"
npm ERR! node v7.10.0
npm ERR! npm  v4.2.0
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! app@1.0.0 test: `eslint src && preact test`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the app@1.0.0 test script 'eslint src && preact test'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the app package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     eslint src && preact test
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs app
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls app
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/oren/.npm/_logs/2017-06-12T00_55_36_627Z-debug.log
```

* npm 4.2.0
* node v7.10.0
* Ubuntu 16.04.2 LTS

### PR DESCRIPTION
**Pull request by developit**
Fixes #112.  /cc @reznord

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
     "build": "babel src -d lib -D",
     "prepublish": "npm run -s build",
     "dev": "babel-node src",
-    "lint": "eslint src tests",
+    "lint": "eslint src tests examples --ignore-pattern build && editorconfig-tools check *.js tests/**.test.js src/**/*.{js,json,html}",
```
Comment (developit) [inline package.json]: @todo windows

**Comment by reznord** [comment]
@developit [here](https://github.com/developit/preact-cli/blob/51e96b509ef03184bacf6f616a952fe45e880c7f/examples/full/src/components/app.js#L11) it is throwing the following error:

`error  Expected line before comment  lines-around-comment`

I guess this is the reason why it is failing in the tests.

**Comment by rkostrzewski** [comment]
@thangngoc89 tests are failing because globs aren't expanded - from what the readme says there should be `globstar` available in the shell

I might be crazy here but why we don't just write js script that uses `node-glob` & `editorconfig-tools` 😆

**Comment by thangngoc89** [comment]
@rkostrzewski Not sure why I were tagged here?

**Comment by rkostrzewski** [comment]
I've messed up 😛

**Comment by developit** [comment]
I've been searching for an npm module that will expand bash-style globs for a command.  I can't find one - anyone know of one?

**Comment by developit** [comment]
If there's a config that matches, sure

**Comment by reznord** [comment]
Do we really need this PR now? Since we have totally isolated the examples into templates.

**Comment by lukeed** [comment]
Don't think so!

**Pull request closed without merging**